piccolo/testing/model_builder.py

```diff
@@ -159,29 +151,110 @@ def _randomize_attribute(cls, column: Column) -> t.Any:
             Column class to randomize.
 
         """
-        random_value: t.Any
-        if column.value_type == Decimal:
-            precision, scale = column._meta.params["digits"] or (4, 2)
-            random_value = RandomBuilder.next_float(
-                maximum=10 ** (precision - scale), scale=scale
-            )
-        elif column.value_type == datetime.datetime:
-            tz_aware = getattr(column, "tz_aware", False)
-            random_value = RandomBuilder.next_datetime(tz_aware=tz_aware)
-        elif column.value_type == list:
-            length = RandomBuilder.next_int(maximum=10)
-            base_type = t.cast(Array, column).base_column.value_type
-            random_value = [
-                cls.__DEFAULT_MAPPER[base_type]() for _ in range(length)
-            ]
-        elif column._meta.choices:
+        reg = cls.get_registry(column)
+        if column._meta.choices:
             random_value = RandomBuilder.next_enum(column._meta.choices)
         else:
-            random_value = cls.__DEFAULT_MAPPER[column.value_type]()
+            random_value = reg[column.value_type]()
 
-        if "length" in column._meta.params and isinstance(random_value, str):
-            return random_value[: column._meta.params["length"]]
-        elif isinstance(column, (JSON, JSONB)):
+        if isinstance(column, (JSON, JSONB)):
             return json.dumps({"value": random_value})
-
         return random_value
+
+    @classmethod
+    def get_registry(
+        cls, column: Column
+    ) -> MappingProxyType[t.Type, t.Callable]:
+        """
+        This serves as the public API allowing users to **view**
+        the complete registry for the specified column.
+
+        :param column:
+            Column class to randomize.
+
+        """
+        default_mapper = cls.__DEFAULT_MAPPER
+        if not default_mapper:  # execute once only
+            for typ, callable_ in RandomBuilder.get_mapper().items():
+                default_mapper[typ] = callable_
+
+        # order matters
+        reg = {
+            **default_mapper,
+            **cls._get_local_mapper(column),
+            **cls._get_other_mapper(column),
+        }
+
+        if column.value_type == list:
+            reg[list] = partial(
+                RandomBuilder.next_list,
+                reg[t.cast(Array, column).base_column.value_type],
```

Comment:
I don't want to make things too insane, but multidimensional arrays are possible:

```python
Array(Array(Integer())
```

I wonder what would happen here in that situation?

Reply:
Thank you for bringing this situation to my attention. I was curious about the behavior in our current codebase, so I conducted a quick test. It seems that the current implementation will throw a `KeyError` for this situation (please correct me if I'm mistaken).

Reply:
This might be a bit off-topic, but I wanted to try out this behavior in the `Piccolo playground`, and it doesn't seem to be working. The code works fine if I just launch a terminal and enter the shell.
```
piccolo playground run --engine=sqlite3
```

```python
In [1]: from piccolo.table import Table

In [2]: from piccolo.columns import Array, BigInt

In [3]: class MyTable(Table):
   ...:     my_column = Array(Array(BigInt()))
   ...:
---------------------------------------------------------------------------
NameError                                 Traceback (most recent call last)
Cell In[3], line 1
----> 1 class MyTable(Table):
      2     my_column = Array(Array(BigInt()))

Cell In[3], line 2, in MyTable()
      1 class MyTable(Table):
----> 2     my_column = Array(Array(BigInt()))

NameError: name 'Array' is not defined
```

Reply:
@jrycw This happens because the `Array` and `BigInt` columns are not imported into the [Playground application](https://github.com/piccolo-orm/piccolo/blob/master/piccolo/apps/playground/commands/run.py#L13-L23). If you patch the local Piccolo installation and add these two columns, everything works fine. Maybe should add all the possible columns to import and that would solve the problem..

Reply:
@sinisaos Thank you very much for identifying the source of the issue and providing the solution. It's working now.

Reply:
You're right that the current implementation of `ModelBuilder` doesn't handle multidimensional arrays, so I wouldn't worry about it if it's a tricky fix. `Array` columns now have two methods to help with this kind of thing: `_get_dimensions` and `_get_inner_value_type`,